dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -4593,11 +4593,26 @@ RtpsUdpDataLink::get_addresses_i(const RepoId& local) const {
   return retval;
 }
 
+bool RtpsUdpDataLink::RemoteInfo::insert_recv_addr(AddrSet& aset) const
+{
+  if (last_recv_addr_ == ACE_INET_Addr()) {
+    return false;
+  }
+  for (AddrSet::iterator it = unicast_addrs_.begin(); it != unicast_addrs_.end(); ++it) {
+    if (it->get_type() == last_recv_addr_.get_type() &&
```

Comment:
Thinking about this a bit more, this probably also needs to check the relay address as well as what's in the locators.

Reply:
Nevermind, you're right... last_recv_addr_ shouldn't ever be set to the relay address based on the logic at the top of update_last_recv_addr, so no need to check for it.